### PR TITLE
Tab bugs and clean-up

### PR DESCRIPTION
**Pull request by bartdominiak**
Refs: #305 #304
internal: dev-37785

### CODE REVIEW

File: components/02-elements/tab/tab.js [hunk 1]
```diff
@@ -7,8 +7,8 @@
         mediaQuery         = window.matchMedia('(min-width: 768px)');
 
   tab.forEach(element => {
-    const title   = element.querySelectorAll('.tab__title'),
-          content = element.querySelectorAll('.tab__content');
+    const title   = element.querySelectorAll(':scope > .tab__title'),
+          content = element.querySelectorAll(':scope > .tab__content');
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: `:scope` doesn't have enough support in browsers
https://developer.mozilla.org/en-US/docs/Web/CSS/:scope

I'd suggest to use something like this
```js
element.children.forEach(child => {
  if (child.classList.contains('tab__title')) {
    // title logic
  }

  if (child.classList.contains('tab__content')) {
    // content logic
  }
})
```

File: components/02-elements/tab/tab.js [hunk 2]
```diff
@@ -3,38 +3,31 @@
 (function() { // eslint-disable-line
   const tab                = [...document.querySelectorAll('.tab')],
         activeTitleClass   = 'tab__title--active',
-        activeContentClass = 'tab__content--active',
-        mediaQuery         = window.matchMedia('(min-width: 768px)');
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: Why you removed all code related to MQ? I'm pretty sure it's required in Alpaca.

File: components/02-elements/tab/tab.js [hunk 3]
```diff
@@ -3,38 +3,31 @@
...
-            tabId = self.dataset.tab;
+    children.forEach(item => {
 
-      if (!self.classList.contains(activeTitleClass)) {
-        title.forEach(key => key.classList.remove(activeTitleClass));
-        self.classList.add(activeTitleClass);
-      }
+      if (item.classList.contains('tab__title')) {
+        item.addEventListener('click', (e) => {
+          const self = e.currentTarget,
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: It's not necessary, you are listening events on one element `item`

File: components/02-elements/tab/tab.js [hunk 4]
```diff
@@ -3,38 +3,31 @@
...
+                tabId = self.dataset.tab;
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: Move outside event callback, it's a static thing.

File: components/02-elements/tab/tab.js [hunk 5]
```diff
@@ -3,38 +3,31 @@
...
 
-    window.addEventListener('resize', () => {
-      if (mediaQuery.matches
-          && !document.querySelectorAll('.tab__title--active').length
-      ) {
-        title[0].classList.add('tab__title--active');
-        content[0].classList.add('tab__content--active');
+              if (key.dataset.content === tabId &&
+                  key.classList.contains('tab__content')) {
+                item.classList.add(activeTitleClass);
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: It shouldn't be in a loop or condition, all you need to know is if this tab was selected before.

File: components/02-elements/tab/tab.js [hunk 6]
```diff
@@ -3,38 +3,31 @@
...
-        if (key.dataset.content === tabId
-            && !key.classList.contains(activeContentClass)
-        ) {
-          content.forEach(key => key.classList.remove(activeContentClass));
-          key.classList.add(activeContentClass);
-        }
-      });
-    }));
+          if (!self.classList.contains(activeTitleClass)) {
+            children.forEach(key => {
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: How is `key` related to DOM elements?

File: components/02-elements/tab/tab.js [hunk 7]
```diff
@@ -3,38 +3,31 @@
...
+              key.classList.remove(activeContentClass);
 
-    window.addEventListener('resize', () => {
-      if (mediaQuery.matches
-          && !document.querySelectorAll('.tab__title--active').length
-      ) {
-        title[0].classList.add('tab__title--active');
-        content[0].classList.add('tab__content--active');
+              if (key.dataset.content === tabId &&
+                  key.classList.contains('tab__content')) {
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: Formatting.

File: components/02-elements/tab/tab.js (hunk 3 above)
Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: Changed.

File: components/02-elements/tab/tab.js (hunk 4 above)
Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: Done.

File: components/02-elements/tab/tab.js (hunk 7 above)
Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: Done.

File: components/02-elements/tab/tab.js (hunk 2 above)
Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: It was added by me, b/c I wanted to add opened first tab on resize, but it's unecessary I think.

File: components/02-elements/tab/tab.js (hunk 6 above)
Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: after click I need to delete `active` tag from other tabs. It's necessary to forEach again. (I think)

File: components/02-elements/tab/tab.js (hunk 5 above)
Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: after click I need to delete `active` tag from other tabs. It's necessary to forEach again. (I think)

Comment (bartdominiak) [reply components/02-elements/tab/tab.js]: and also if `dataset` match then I add `active` tag to content

**Pull request merged**